cloud_AWS/terraform/module/examples/multiple-vpc-regions/aws_multiregions.py

```diff
@@ -0,0 +1,174 @@
+import argparse
+import logging
+import json
+import os
+import sys
+from typing import Any, Callable, Dict, Tuple
+
+from python_terraform import IsFlagged, IsNotFlagged, Terraform
+
+log = logging.getLogger(__name__)
+logging.basicConfig(
+    format="%(asctime)s [%(name)s] %(levelname)s: %(message)s", level=logging.INFO, filename="multiregions.log"
+)
+
+EX_OK: int = 0  # exit code for successful command
+EX_FAILED: int = 1  # exit  code for failed command
+
+DFAULT_VPC_REGIONS_FILE_NAME: str = "input_data.json"
+
+TerraformVars = Dict[str, Any]  # variables passed in terraform plan/apply/destroy call
+TerraformAction = Callable[[Terraform, TerraformVars], bool]  # terraform plan/apply/destroy
+
+
+def execute_action(action: TerraformAction, vpc_regions: dict) -> bool:
+    """Execute an action for every region"""
+
+    t_bucket = Terraform(working_dir="bucket/")
+    bucket_region_name = []
+    bucket_arn_list = []
+    err_occur = False
+
+    # invoke action for each region on terraform configuration located in bucket/
+    for region in vpc_regions["regions"]:
+        print_log(f'Region: {region}, vpc ids: {vpc_regions["regions"][region]["vpc_id_list"]}')
+        tf_vars = {
+            "region": region,
+            "vpc_id_list": vpc_regions["regions"][region]["vpc_id_list"],
+            "s3_bucket_prefix": vpc_regions["regions"][region].get("s3_bucket_prefix", "")
+        }
+
+        if not (prepare_workspace(t_bucket, region) and action(t_bucket, tf_vars)):
+            err_occur = True
```

Comment:
Not sure, should we continue if we had an error? t_bucket.output() may be undefined I guess

Reply:
If an error occurs, for example in one region, the script applies the terraform plan to the other regions. If bucket.output() is empty, terraform will execute the cloud_iam module, but with empty buckets.
I cannot conditionally execute cloud_iam based on bucket.output() values because the output in plan/destroy actions do not contain the required values.

Reply:
Resolved to return False on error